api/search.go

```diff
@@ -132,3 +140,37 @@ func SearchHandlerFunc(queryBuilder QueryBuilder, elasticSearchClient ElasticSea
 
 	}
 }
+
+func CreateSearchIndexHandlerFunc(elasticSearchClient ElasticSearcher) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		ctx := req.Context()
+		indexName := createIndexName("ons")
+		fmt.Printf("Index created: %s\n", indexName)
+
+		indexCreated, err := elasticSearchClient.CreateNewEmptyIndex(ctx, indexName)
+		if !indexCreated {
+			if err != nil {
+				http.Error(w, serverErrorMessage, http.StatusInternalServerError)
+			}
+			return
```

Comment:
Im not sure about this logic. If the index is not created and there is no error should this return a 200 status? I'd argue no as the request has failed and we should be conveying this to the caller otherwise its a silent failure. I am not sure what the appropriate response status should be for this case though.

Reply:
Yes, I see what you mean. I've change it so that it will always throw an internal server error if the index does not get created now. And if err is not null then it will log that instead of returning at that point.